HIP/hip-729.md

```diff
@@ -0,0 +1,129 @@
+---
+hip: 729
+title: Contract Accounts Nonce Externalization
+author: Miroslav Gatsanoga <miroslav.gatsanoga@limechain.tech>
+working-group: Stoyan Panayotov <stoyan.panayotov@limechain.tech>, Ivan Kavaldzhiev <ivan.kavaldzhiev@limechain.tech>, Richard Bair <@rbair23>, Nana Essilfie-Conduah <@Nana-EC>
+type: Standards Track
+category: Service
+needs-council-approval: Yes
+status: Review
+created: 2023-05-05
+discussions-to: https://github.com/hashgraph/hedera-improvement-proposal/discussions/730
+updated: 2023-05-16
+---
+
+## Abstract
+
+The Hedera Smart Contract Service (HSCS) supports the deployment of contracts. A deployed contract `A` should have a nonce value that reflects the number of other contracts that were created since `A`'s creation.
+Currently this is not the observed behavior in Hedera and this HIP aims to fix that.
+
+## Motivation
+
+In Ethereum, contract accounts have nonces, as specified by the [yellow paper](http://gavwood.com/paper.pdf) and [EIP-161](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-161.md).
+
+The EVM uses the nonce value internally, for several different operations, for example:
+    1. to check whether an account exists at an address, it checks whether the nonce property stored for this address is different than 0 (it also checks the balance and code properties for the address as well, but this is out of scope for this discussion).
+    2. when deploying new contracts via a `CREATE` operation, the address for an Ethereum contract is **deterministically** computed from the address of its creator (`sender`) and how many transactions the creator has sent (`nonce`). The `sender` and `nonce` are RLP encoded and then hashed with Keccak-256.
+    3. when deploying new contracts via a `CREATE2` operation, the derivation of new contract addresses does not involve the nonce. Instead, a contract address can be derived using the deployer address, bytecode, and salt.
```

Comment:
nit: having it as a 3rd point makes it read as if the nonce is utilized in `CREATE2`.
The content is good but i'd make a separate sentence right after number 2 instead of number 3

Reply:
Updated.